src/pages/index.js

```diff
@@ -41,62 +182,136 @@ editProfileMOB.addEventListener("click", () => {
+      editProfilePopup.popupForm.reset();
+      editProfileFormValidator.toggleButtonState();
+      editProfilePopup.close();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      submitButton.textContent = "Save";
+    });
 }
 
+//Handles new card submit
 function handleAddCardSubmit(cardData) {
-  createCard(cardData, "prepend");
-  addPlacePopup.popupForm.reset();
-  addPlaceFormValidator.toggleButtonState();
-  addPlacePopup.close();
+  const submitButton = addCardModal.querySelector(
+    "#add-card-modal-submit-button"
+  );
+  submitButton.textContent = "Saving...";
+
+  api
+    .addNewCard(cardData)
+    .then((res) => {
+      addNewCard(res, "prepend");
+      addPlacePopup.popupForm.reset();
+      addPlaceFormValidator.toggleButtonState();
+      addPlacePopup.close();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      submitButton.textContent = "Create";
+    });
 }
 
-const handleImageClick = (cardData) => {
-  maxImagePopup.open(cardData);
-};
+//Handles card deletion
+let cardtoDeleteData;
 
-//***************************************************************************************************************************//
-//                                                     Objects                                                               //
-//***************************************************************************************************************************//
+//Handles confirm delete and sets delete data
+function handleConfirmDelete(data) {
+  confirmDeletePopup.open();
+  cardtoDeleteData = data;
+}
 
-const cardSection = new Section(initialCards, createCard, "#cards-list");
-const maxImagePopup = new PopupWithImage("#max-image-modal");
-const editProfilePopup = new PopupWithForm(
-  "#edit-profile-modal",
-  handleEditProfile
-);
-const addPlacePopup = new PopupWithForm("#add-card-modal", handleAddCardSubmit);
+//Deletes card
+async function handleDeleteCard() {
+  try {
+    api.deleteCard(cardtoDeleteData._id);
+    const card = document.querySelector(`li[id='${cardtoDeleteData._id}'`);
+    card.remove();
+  } catch (err) {
+    console.log(err);
+  } finally {
+    confirmDeletePopup.close();
+  }
+}
 
-const userInfo = new UserInfo({
-  nameSelector: ".profile__name",
-  jobSelector: ".profile__subtitle",
-});
+//Changes Profile Picture
+async function handleEditPicSubmit({ link }) {
+  const submitButton = editProfilePicModal.querySelector(
+    "#edit-pic-modal-submit-button"
+  );
+  submitButton.textContent = "Saving...";
 
-//***************************************************************************************************************************//
-//                                             Validation and Rendering                                                      //
-//***************************************************************************************************************************//
+  try {
```

Comment:
Please, don't use `try/catch` if you don't know how to handle server responses. You need to wait for the server requests

It's better to use `then...catch...finally` rather than `try/catch` to make it easier for you